tsa/pdf_to_json/main.py

```diff
@@ -1,14 +1,60 @@
 import base64
 import json
+import io
 
 from fastapi import FastAPI, Request
 from google.cloud import storage
+import pdfplumber
+
+from tables.manager import TableManager
 
-from model import pdf_to_json
 
 app = FastAPI()
 
 
+def pdf_file_like(bucket_name, blob_name):
+    client = storage.Client()
+    bucket = client.bucket(bucket_name)
+    blob = bucket.get_blob(blob_name)
+    pdf_bytes = blob.download_as_bytes()
+    pdf_file_like = io.BytesIO(pdf_bytes)
+    return pdf_file_like
+
+
+def table_to_json(columns, rows):
+    json_data = []
+    previous_values = {column: None for column in columns}
+    for row in rows:
+        row = dict(zip(columns, row))
+        for key, value in row.items():
+            if value is None:
+                row[key] = previous_values[key]
+            else:
+                previous_values[key] = value
+        json_data.append(row)
+    return json_data
+
+
+def clean_name(column):
```

Comment:
listcomp?

```
def clean_name(column):
    return ''.join(char for char in column if char not in "\n ").lower()
```

Seems you can just do `not in "\n "` without the split()?

Reply:
yeah, think that should work. Thanks.